addons/l10n_it_edi_extended/models/account_move.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from collections import namedtuple
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _l10n_it_edi_fatturapa_template(self, is_simplified=False):
+        if not is_simplified:
+            return 'l10n_it_edi_extended.account_invoice_it_FatturaPA_export_extended'
+        return super()._l10n_it_edi_fatturapa_template(is_simplified)
+
+    def _l10n_it_edi_filter_fatturapa_tax_details(self, line):
+        return (line['tax_repartition_line_id'].factor_percent >= 0
+                and line['tax_id'].amount >= 0
+                and not line['tax_id'].l10n_it_pension_fund_type)
+
+    def _l10n_it_edi_extended_get_tax_kind(self, tax):
+        return ((tax.l10n_it_withholding_type and 'withholding')
+                or (tax.l10n_it_pension_fund_type and 'pension_fund')
+                or 'vat')
+
+    def _l10n_it_edi_fatturapa_extended_values(self, template_values):
+        """
+            Add withholding and pension_fund features.
+        """
+        template_values = super()._l10n_it_edi_fatturapa_extended_values(template_values)
+
+        no_tax = self.env['account.tax']
```

Comment:
We should be able to get these values immediately from the tax line.  invoice.line_ids.filtered(lambda l: l.tax_line_id and l.tax_line_id.l10n_it_pension_fund) e.g. and that should be one tax for the total.

Reply:
Not for the base amount. The base amount has to be summed up.

Reply:
There is a field tax_base_amount as well.  Although, they want to remove it maybe, in here it is super in handy.

Reply:
That's in the tax details, not on the lines, and it's divided by line, so it's to be summed up again

Reply:
https://github.com/odoo/odoo/blob/fcad4dbdbd65775b75aa94049da06bd0b07295d8/addons/account/models/account_move.py#L3674